boudicca.base/enricher/src/main/kotlin/base/boudicca/enricher/service/osm/OsmEnricher.kt

```diff
@@ -0,0 +1,218 @@
+) : Enricher {
+    val sourceVariant = listOf(Variant(SOURCE_VARIANT_NAME, "OSM"))
+    val logger = LoggerFactory.getLogger(this::class.java)!!
+
+    val mapper = jacksonObjectMapper()
+
+    override fun enrich(event: StructuredEvent): StructuredEvent {
+        val osmId: String? = getOrSearchOsmId(event)
+        if (osmId == null || !isValidOsmIdFormat(osmId)) {
+            return event
+        }
+        val enrichedEvent = enrichEventFromOverpass(osmId, event)
+
+        return enrichedEvent
+    }
+
+    @Suppress("detekt.LongMethod")
+    private fun enrichEventFromOverpass(
+        osmId: String,
+        event: StructuredEvent
+    ): StructuredEvent {
+        val url = "$NOMINATIM_BASE_URL/lookup?format=jsonv2&extratags=1&namedetails=1&osm_ids=${osmId}"
+        val response = synchronized(fetcher) {
+            fetcher.fetchUrl(url)
+        }
+
+        // parse osm data into our properties
+        val builder = event.toBuilder()
+        val json = mapper.readTree(response)
+        if (json.isEmpty) {
+            logger.warn("nominatim did not return any results for osm id $osmId")
+            return event
+        }
+        val firstElement = json.first()
+        val tags = firstElement["extratags"]
+        val address = firstElement["address"]
+
+        val currentSources = ListFormat.parseFromString(event.data[SemanticKeys.SOURCES_PROPERTY.getKey()])
+        builder.withProperty(
+            property = SemanticKeys.SOURCES_PROPERTY,
+            value = currentSources + "Source:OSM:${firstElement["licence"]?.asText()}"
+        )
+
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_COORDINATES_LAT_PROPERTY,
+            firstElement["lat"]?.asDouble(),
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_COORDINATES_LON_PROPERTY,
+            firstElement["lon"].asDouble(),
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_NAME_PROPERTY,
+            firstElement["name"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_DESCRIPTION_PROPERTY,
+            firstElement["category"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_OSM_ID_PROPERTY,
+            osmId
+        )
+        if (address != null) {
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_ADDRESS_PROPERTY,
+                buildAddress(address)
+            )
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_CITY_PROPERTY,
+                address["city"]?.asText()
+            )
+        }
+        if (tags != null) {
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_CONTACT_EMAIL_PROPERTY,
```

Comment:
Would it make sense to put all the boudicca tags values and the osm tags values in a common configuration class? so it is easier to maintain

could be a simple map<String, Property>

Then it could be intuitively written as

(osm Tag) "wikidata" (maps) to SemanticKeys.LOCATION_WIKIDATA_PROPERTY
contacts:website to SemanticKeys.LOCATION_URL_PROPERTY
...

Reply:
sadly nominatim has it's own data format which changes and keeps osm tags, depending on what the tag is. as you can see we take properties from different json objects here, so a common map does not really work or make it clearer/more reusable